### DISCUSSION TITLE
monaco + bundled vscode extension in one project

### DISCUSSION THREAD

**Discussion by cdietrich** [eclipse-langium/langium#887] [Q&A]
hi,
is there any working example with monaco + bundled vscode extension in one project
in the minilogo example i cannot get it running
thanks

**Comment by msujew** [comment]
@cdietrich Yep, seems like something went wrong when updating to the newest version of the monaco wrapper component. I've created [a PR to fix this](https://github.com/langium/langium-minilogo/pull/5). You can use that branch to run everything on your own by using `npm run build:web` and `npm run serve`.

**Reply by cdietrich** [reply]
i had got this already running in my example.
the problem is the vscode extension: if i install it and open examples it does not seems to work.

**Reply by cdietrich** [reply]
e.g.
```
git clean -fdx && npm install && npm run langium:generate && vsce package
```
results in
```
 ERROR  Extension entrypoint(s) missing. Make sure these files exist and aren't ignored by '.vscodeignore':
  extension/out/extension.js

```

**Reply by cdietrich** [reply]
and just using main.js does not seems to help

**Reply by cdietrich** [reply]
the main.js has "Qp.join("out","language-server","main")"

**Reply by msujew** [reply]
You're right, I've pushed another commit to the PR that also addresses that issue. The build scripts didn't generate the extension into the correct bundle and also didn't bundle the language-server at all.

**Reply by cdietrich** [reply]
maybe we need something like
```
"esbuild-base": "esbuild ./src/extension.ts --bundle --outfile=out/extension.js --external:vscode --format=cjs --platform=node && esbuild ./src/language-server/main.ts --bundle --outfile=out/language-server/main.js --external:vscode --format=cjs --platform=node",
```
